### ISSUE TITLE
Duplicate code (part 3)

### ISSUE THREAD

**Issue by LucasSte** [hyperledger/solang#1263]
1.

https://github.com/hyperledger/solang/blob/c5e430461b9f9934d08afa2fc3b81962637c6c5d/src/emit/math.rs#L90-L107

https://github.com/hyperledger/solang/blob/c5e430461b9f9934d08afa2fc3b81962637c6c5d/src/emit/math.rs#L318-L335

===============================================

2.

https://github.com/hyperledger/solang/blob/c5e430461b9f9934d08afa2fc3b81962637c6c5d/src/sema/expression/member_access.rs#L419-L427

https://github.com/hyperledger/solang/blob/c5e430461b9f9934d08afa2fc3b81962637c6c5d/src/sema/expression/member_access.rs#L431-L439

===============================================

3.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L421-L437

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L455-L471

===============================================

4.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L484-L500

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L620-L636

===============================================

5.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L513-L543

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L618-L648

===============================================

6.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L583-L611

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L654-L682

===============================================

7.
https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L908-L934

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L940-L966

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L971-L997

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/sema/expression/mod.rs#L1050-L1076

===============================================
8.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/mod.rs#L988-L1004

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/mod.rs#L1010-L1026

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/mod.rs#L1184-L1201

===============================================
9.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/mod.rs#L1600-L1613

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/mod.rs#L1623-L1634

===============================================
10.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/encoding/mod.rs#L370-L375

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/encoding/mod.rs#L501-L510

===============================================
11.
https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/reaching_definitions.rs#L213-L219

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/codegen/vector_to_slice.rs#L155-L161

===============================================
12.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/bin/solang.rs#L84-L99

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/bin/solang.rs#L231-L246

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/bin/solang.rs#L277-L292

===============================================
13.

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/bin/solang.rs#L129-L146

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/bin/solang.rs#L247-L264

https://github.com/hyperledger/solang/blob/1b46d73b32449a09ba1c1e54c8073ecb4d7cf4a0/src/bin/solang.rs#L293-L310

**Comment by Celetert** [comment]
Hello! I am a beginner in FoSS ecosystem. I want to ask that wheather i should check issues those are not yet assigned only?

**Comment by xermicus** [comment]
> Hello! I am a beginner in FoSS ecosystem. I want to ask that wheather i should check issues those are not yet assigned only?

Yes, the general idea is that if someone is assigned to an issue, that issue is being worked on. If you spot a stale issue (e.g. the issue is assigned but there were no discussion and commits in a related PR happening  for weeks), feel free to ping the assignee whether they are still working on it or not.

I can assign this issue here to you if you want?

**Comment by anj20** [comment]
@LucasSte  can u assign this to me?

**Comment by LucasSte** [comment]
> @LucasSte can u assign this to me?

Done!

**Comment by LucasSte** [comment]
> @LucasSte can u assign this to me?

Hey @anj20, you were assigned to this issue two weeks ago, but we have not heard from you since then. Are you still interested in working on this?
If not, we better deassign you, so others can have the opportunity to contribute.

**Comment by startup-dreamer** [comment]
I would like to work on this

**Comment by seanyoung** [comment]
@startup-dreamer looking forward to seeing a pull request

**Comment by startup-dreamer** [comment]
I am little busy on personal project if anyone wants to take this he can, I will start work as soon as project is completed.

**Comment by startup-dreamer** [comment]
whenever i am trying to run `cargo test -p solang --test solana`
it is giving me the error 
``` 
     error: No suitable version of LLVM was found system-wide or pointed
              to by LLVM_SYS_150_PREFIX
              Consider using `llvmenv` to compile an appropriate copy of LLVM, and
              refer to the llvm-sys documentation for more information.
```
I have installed llvmenv (latest version) and set environmental variable LLVM_SYS_150_PREFIX as `export LLVM_SYS_150_PREFIX=$(llvmenv prefix)`
still it is not working any idea how to resolve this.

**Comment by seanyoung** [comment]
@startup-dreamer please follow the instructions at https://solang.readthedocs.io/en/latest/installing.html#step-1-install-the-llvm-libraries and do a `cargo clean`

**Comment by startup-dreamer** [comment]
tried but error is persisting i don't think i will be able to resolve this issue sorry for the trouble
